scripts/run_integration_tests.sh

```diff
@@ -78,24 +78,40 @@ popd
 
 # Build custom kernel based on virtio-pmem and virtio-fs upstream patches
 VMLINUX_IMAGE="$WORKLOADS_DIR/vmlinux"
+VMLINUX_PVH_IMAGE="$WORKLOADS_DIR/vmlinux.pvh"
 BZIMAGE_IMAGE="$WORKLOADS_DIR/bzImage"
 
-LINUX_CUSTOM_DIR="linux-custom"
+LINUX_CUSTOM_DIR="$WORKLOADS_DIR/linux-custom"
 
-if [ ! -f "$VMLINUX_IMAGE" ]; then
+if [ ! -f "$VMLINUX_IMAGE" -o ! -f "$VMLINUX_PVH_IMAGE" ]; then
     SRCDIR=$PWD
     pushd $WORKLOADS_DIR
     time git clone --depth 1 "https://github.com/cloud-hypervisor/linux.git" -b "virtio-fs-virtio-iommu-topo-5.5-rc1" $LINUX_CUSTOM_DIR
+    cp $SRCDIR/resources/linux-virtio-fs-virtio-iommu-config $LINUX_CUSTOM_DIR/.config
+    popd
+fi
+
+if [ ! -f "$VMLINUX_IMAGE" ]; then
     pushd $LINUX_CUSTOM_DIR
-    cp $SRCDIR/resources/linux-virtio-fs-virtio-iommu-config .config
+    scripts/config --disable "CONFIG_PVH"
```

Comment:
I would call `cp $SRCDIR/resources/linux-virtio-fs-virtio-iommu-config $LINUX_CUSTOM_DIR/.config` from here, this way we don't need to disable `CONFIG_PVH`.

Reply:
I'm not sure I understand the request but let me try and clarify my approach.

`CONFIG_PVH` is already enabled by default in upstream Linux config, including the one provided by ICH in `resources/linux-virtio-fs-virtio-iommu-config`. This is unlikely to change as PVH is a stable interface. But just in case that a config file is used that has different defaults, I opted for explicitly enabling and disabling the option in the script so there is no reliance on default values that may or may not change in the future.

Using the "insider knowledge" that PVH is enabled by default in the config, we could do the `VMLINUX_PVH_IMAGE` build first, and then disable the option for the `VMLINUX_IMAGE` build. But all we would be saving is one invocation of `scripts/config` which is a text substitution (AFAICT `scripts/config` does not do any dependency checking) and should be pretty fast anyways.

Does my reasoning above make sense?

Reply:
Oh yes I understand now, I wasn't aware PVH was enabled by default already. Thanks for the clear explanation.